internal/pkg/agent/application/application.go

```diff
@@ -159,6 +161,11 @@ func mergeFleetConfig(rawConfig *config.Config) (storage.Store, *configuration.C
 			errors.M(errors.MetaKeyPath, path))
 	}
 
+	// Fix up fleet.agent.id otherwise the fleet.agent.id is empty string
```

Comment:
Why is this necessary?

Reply:
Was working off of the endpoint fleet config example and the ```fleet.agent.id``` was empty from the fleet configuration file. The agent.id was at top level but not under the ```fleet``` which we suppose to pass to the Endpoint.

```
fleet:
  agent:
    id: 01010101-0101-0101-0101-010101010101
    logging:
      level: debug
  host:
    id: dabadaba-d000-0000-0000-000000000000
  access_api_key: akjdgskjhdgskhgdskhfgshjdfgsjhdgdfsjhdgfsjsdfjhgsdfagjshUQ==
  protocol: https
  hosts:
  - localhost:5601
  path: /boo
  proxy_disable: false
  proxy_url: ""
  ssl:
    verification_mode: full
    ca_trusted_fingerprint: 96BCEC06264976F37460779ACF28C5A7CFE8A3C0AAE11A8FFCEE05C0BDDF08C6
    certificate_authorities:
    - /tmp/dan/ca_certs
    certificate: "/etc/pki/client/cert.pem"
    key: "/etc/pki/client/cert.key"
    ca_sha256: 7HIpactkIAq2Y49orFOOQKurWxmmSFZhBCoQYcRhJ3Y=
```